pilot/cmd/pilot-agent/options/agent.go

```diff
@@ -71,6 +73,9 @@ func NewAgentOptions(proxy *model.Proxy, cfg *meshconfig.ProxyConfig) *istioagen
 		DualStack:                   features.EnableDualStack,
 		UseExternalWorkloadSDS:      useExternalWorkloadSDSEnv,
 		MetadataDiscovery:           enableWDSEnv,
+		SDSFactory: func(options *security.Options, workloadSecretCache security.SecretManager, pkpConf *meshconfig.PrivateKeyProvider) istioagent.SDSService {
```

Comment:
How does this help reduce imports? We still import `sds` and would always do so, even if the underlying sds.newServer is later changed to import less

Reply:
It's temporary so we can lock down all of `pkg/istio-agent`. I don't know when we can convert SDS to be slim. Once that is done, we can simplify. CC @keithmattix

Reply:
Can we get this in while we sort out SDS?